tests/conftest.py

```diff
@@ -610,28 +598,36 @@ def get_host_data(request, dut):
     return get_host_vars(inv_files, dut)
 
 
-def generate_params_frontend_hostname(request, duts_in_testbed, tbname):
+def generate_params_frontend_hostname(request):
     frontend_duts = []
+    tbname, tbinfo = get_tbinfo(request)
+    duts = tbinfo['duts']
     inv_files = get_inventory_files(request)
-    for dut in duts_in_testbed:
+    for dut in duts:
         if is_frontend_node(inv_files, dut):
             frontend_duts.append(dut)
     assert len(frontend_duts) > 0, \
         "Test selected require at-least one frontend node, " \
-        "none of the DUTs '{}' in testbed '{}' are a supervisor node".format(duts_in_testbed, tbname)
+        "none of the DUTs '{}' in testbed '{}' are a supervisor node".format(duts, tbname)
     return frontend_duts
 
 
-def generate_params_frontend_hostname_rand_per_hwsku(request, duts_in_testbed, tbname):
-    frontend_hosts = generate_params_frontend_hostname(request, duts_in_testbed, tbname)
+def generate_params_hostname_rand_per_hwsku(request, frontend_only=False):
+    tbname, tbinfo = get_tbinfo(request)
+    hosts = tbinfo['duts']
+    if frontend_only:
+        hosts = generate_params_frontend_hostname(request)
     inv_files = get_inventory_files(request)
     # Create a list of hosts per hwsku
     host_hwskus = {}
-    for a_host in frontend_hosts:
-        a_host_hwsku = get_host_visible_vars(inv_files, a_host, variable='hwsku')
-        if not a_host_hwsku:
+    for a_host in hosts:
+        host_vars = get_host_visible_vars(inv_files, a_host)
+        if 'hwsku' in host_vars:
+            a_host_hwsku = host_vars['hwsku']
+        else:
             # Lets try 'sonic_hwsku' as well
-            a_host_hwsku = get_host_visible_vars(inv_files, a_host, variable='sonic_hwsku')
+            if 'sonic_hwsku' in host_vars:
+                a_host_hwsku = host_vars['sonic_hwsku']
```

Comment:
If there is neither 'hwsku' nor 'sonic_hwsku', then variable `a_host_hwsku` would be undefined. Line 631 would raise exception. It needs to be initialized in advance.

Reply:
Added commit to initialize the variable to None